jbpm-services/jbpm-kie-services/src/test/java/org/jbpm/kie/services/test/BPMN2DataServiceImplMultiThreadBytemanTest.java

```diff
@@ -1,231 +0,0 @@
-        final List<ProcessDefinition> defs = new ArrayList<ProcessDefinition>();
-
-        byte[] process1 = IoUtils.readBytesFromInputStream(this.getClass().getResourceAsStream(
-                "/repo/processes/general/customtask.bpmn"));
-        final String process1Content = new String(process1, "UTF-8");
-        final String SCRIPT_BPMN_PATH = "repo/processes/general/hello.bpmn";
-
-        final CyclicBarrier threadsFinishedBarrier = new CyclicBarrier(3);
-
-        Thread dataServiceThread = new Thread(new Runnable() {
-
-            @Override
-            public void run() {
-                try {
-                    ProcessDefinition def = bpmn2Service.buildProcessDefinition("test", process1Content, null, true);
-                    defs.add(def);
-                } finally {
-                    waitForTheOtherThreads(threadsFinishedBarrier);
-                }
-            }
-        });
-
-        final Exception [] exceptionHolder = new Exception[1];
-
-        final KnowledgeBuilder [] kbuilderHolder = new KnowledgeBuilder[1];
-        Thread kieBuilderThread = new Thread(new Runnable() {
-
-            @Override
-            public void run() {
-
-                try {
-                    kbuilderHolder[0]
-                        = BPMN2DataServiceImplMultiThreadBytemanTest.createKBuilder(SCRIPT_BPMN_PATH, ResourceType.BPMN2);
-                } catch( Exception e ) {
-                    e.printStackTrace();
-                    exceptionHolder[0] = e;
-                } finally {
-                    letDataServiceThreadFinish();
-                    waitForTheOtherThreads(threadsFinishedBarrier);
-                }
-            }
-        });
-
-        String dataServiceThreadName = "dataService";
-        dataServiceThread.setName(dataServiceThreadName);
-        dataServiceThread.start();
-        kieBuilderThread.setName("kieBuilder");
-        kieBuilderThread.start();
-
-        waitForTheOtherThreads(threadsFinishedBarrier);
-
-        assertEquals(1, defs.size());
-
-        if( exceptionHolder[0] != null ) {
-            fail( "See stacktrace: unable to add resource to KieBuilder: " + exceptionHolder[0].getMessage() );
-        }
-
-        Set<String> dataServiceModuleThreads = testItemDefinitionHandler.getUsedByThreads();
-        assertEquals( "Only 1 thread should have accessed the data service handlers!",
-                      1, dataServiceModuleThreads.size() );
-        assertEquals( "Only the data service thread should have accessed the data service handlers!",
-                      dataServiceThreadName, dataServiceModuleThreads.iterator().next() );
-
-        // This will throw an exception if the process instance has not been created well
-        Map<String, Object> params = new HashMap<String, Object>(1);
-        params.put("name", "Poppy");
-        kbuilderHolder[0].newKnowledgeBase().newKieSession().startProcess("hello", params);
```

Comment:
I think we need tests that test not only to make sure that any process built by the data services can also be executed.